### PR TITLE
add steps in tests

### PR DESCRIPTION
**Pull request by YuraSmoliy**

### CODE REVIEW

**Review by andr0306 — APPROVED** [review]

File: src/main/java/pageobjects/HomePageObject.java [hunk 1]
```diff
@@ -61,17 +62,18 @@ public HomePageObject addIphoneToShoppingCartSikuly() {
             screen.wait(scroll, 20);
             screen.type(Key.PAGE_DOWN);
             screen.find(iphoneAddToCartButton).click();
-        } catch (Exception ex) {
+        } catch (org.sikuli.script.FindFailed ex) {
```
Comment (OrestZhmurkevych) [inline src/main/java/pageobjects/HomePageObject.java:65]: Can you just write FindFailed here without explicit library path?

File: src/main/java/pageobjects/HomePageObject.java [hunk 2]
```diff
@@ -61,17 +62,18 @@ public HomePageObject addIphoneToShoppingCartSikuly() {
...
     }
-    public ShoppingCartPageObject openShoppingCartSikuly(){
+
+    public ShoppingCartPageObject openShoppingCartSikuly() {
         openShoppingCartButton = new Pattern(ShoppingCartPatterns.OPEN_SHOPPIN_CART_BUTTON);
         try {
-        screen.wait(openShoppingCartButton, 20).click();
-        } catch (Exception ex) {
+            screen.wait(openShoppingCartButton, 20).click();
+        } catch (org.sikuli.script.FindFailed ex) {
```
Comment (OrestZhmurkevych) [inline src/main/java/pageobjects/HomePageObject.java:76]: Same here

File: src/main/java/pageobjects/ShoppingCartPageObject.java [hunk 3]
```diff
@@ -242,7 +242,7 @@ public Match finedElementInShoppingCartSikuly(Pattern pattern) {
         Match match = null;
         try {
             match = screen.wait(pattern, 20);
-        } catch (Exception ex) {
+        } catch (org.sikuli.script.FindFailed ex) {
```
Comment (OrestZhmurkevych) [inline src/main/java/pageobjects/ShoppingCartPageObject.java:245]: And here

File: src/main/java/pageobjects/ShoppingCartPageObject.java [hunk 4]
```diff
@@ -260,7 +260,7 @@ public ShoppingCartPageObject changeQuantityProductsSikuly() {
             keyboard.type("2");
             screen.find(quantityUpdate).click();
             screen.wait(message, 20);
-        } catch (Exception ex) {
+        } catch (org.sikuli.script.FindFailed ex) {
```
Comment (OrestZhmurkevych) [inline src/main/java/pageobjects/ShoppingCartPageObject.java:263]: And also here)

File: src/main/java/pageobjects/ShoppingCartPageObject.java [hunk 5]
```diff
@@ -273,7 +273,7 @@ public String removeProductSikuly() {
         try {
             screen.wait(quantityForm.targetOffset(50, 0), 20).click();
             messageCart = screen.wait(messageEmptyCart, 20).text();
-        } catch (Exception ex) {
+        } catch (org.sikuli.script.FindFailed ex) {
```
Comment (OrestZhmurkevych) [inline src/main/java/pageobjects/ShoppingCartPageObject.java:276]: And here again)

File: src/main/java/pageobjects/ShoppingCartPageObject.java [hunk 6]
```diff
@@ -284,7 +284,7 @@ public String getTotalCostSikuly() {
         try {
             screen.wait(totalCostProduct, 20);
             text = screen.find(ShoppingCartPatterns.TEXT).text();
-        } catch (Exception ex) {
+        } catch (org.sikuli.script.FindFailed ex) {
```
Comment (OrestZhmurkevych) [inline src/main/java/pageobjects/ShoppingCartPageObject.java:287]: And here too)))

File: src/test/javatest/pageobjectstest/ShoppingCartPageObjectTest.java [hunk 7]
```diff
@@ -64,169 +98,575 @@ public void testAddProductToShoppingCartUseSikuli() {
...
+     * Expected Result: The coupon discount field did not appear when calculating the total value.
+     */
+    @DataProvider()
     public Object[][] createData(Method m) {
-        return new Object[][]{new Object[]{"a111"}, new Object[]{""}, new Object[]{"1111"}};
+        return TestData.INVALID_DATA_FOR_TEST_COUPON;
     }
 
-    @Test(dataProvider = "invalidData", expectedExceptions = org.openqa.selenium.NoSuchElementException.class)
+    @Test(dataProvider = "createData", expectedExceptions = org.openqa.selenium.NoSuchElementException.class)
```
Comment (OrestZhmurkevych) [inline src/test/javatest/pageobjectstest/ShoppingCartPageObjectTest.java:277]: I'm almost sure that you can write just class name without library path

File: src/test/javatest/pageobjectstest/ShoppingCartPageObjectTest.java [hunk 8]
```diff
@@ -64,169 +98,575 @@ public void testAddProductToShoppingCartUseSikuli() {
...
+     * Expected Result: The gift certificate discount field did not appear when calculating the total value.
+     */
+    @DataProvider()
     public Object[][] createDataForGiftCertificate(Method m) {
-        return new Object[][]{new Object[]{"a333"}, new Object[]{""}, new Object[]{"333a"}};
+        return TestData.INVALID_DATA_FOR_TEST_GIFT_CERTIFICATE;
     }
 
-    @Test(dataProvider = "invalidDataGift", expectedExceptions = org.openqa.selenium.NoSuchElementException.class)
+    @Test(dataProvider = "createDataForGiftCertificate", expectedExceptions = org.openqa.selenium.NoSuchElementException.class)
```
Comment (OrestZhmurkevych) [inline src/test/javatest/pageobjectstest/ShoppingCartPageObjectTest.java:361]: Same

**Review by OrestZhmurkevych — APPROVED** [review]

**Review by OrysiaBen — APPROVED** [review]

**Pull request merged**